.github/workflows/python-test.yml

```
@@ -0,0 +1,28 @@
+name: Python tests
+
+on:
+  pull_request:
+    branches: ["main"]
+  push:
+
+permissions:
+  contents: read
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python 3.13
+        uses: actions/setup-python@v3
```

```suggestion
      - uses: actions/setup-python@v3
```
I would also drop that as I cannot be bothered to also update that when updating python 😅